## Examen de Muestra
1. Write a Python program to load the iris data from a given csv file into a dataframe and print the shape of the data, type of the data and first 3 rows. ¿A cuál especie pertenecen estos 3 renglones?

In [251]:
import pandas as pd
import sklearn

In [377]:
from sklearn.datasets import load_iris
iris_dataframe = pd.DataFrame(X_train, columns=iris_dataset.feature_names)
iris_dataframe

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                  5.9               3.0                4.2               1.5
1                  5.8               2.6                4.0               1.2
2                  6.8               3.0                5.5               2.1
3                  4.7               3.2                1.3               0.2
4                  6.9               3.1                5.1               2.3
..                 ...               ...                ...               ...
107                4.9               3.1                1.5               0.1
108                6.3               2.9                5.6               1.8
109                5.8               2.7                4.1               1.0
110                7.7               3.8                6.7               2.2
111                4.6               3.2                1.4               0.2

[112 rows x 4 columns]

In [381]:
print('Shape :' ,iris_dataframe.shape)
print('First three rows : ', iris_dataframe[:3])
print('Type :', type(iris_dataframe))

Shape : (112, 4)
First three rows :     sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.9               3.0                4.2               1.5
1                5.8               2.6                4.0               1.2
2                6.8               3.0                5.5               2.1
Type : <class 'pandas.core.frame.DataFrame'>


In [387]:
#para cargar el archivo como .csv
iris_dataframe_csv = pd.read_csv('iris.csv')
iris_dataframe_csv = iris_dataframe_csv["sepal.length;sepal.width;petal.length;petal.width;variety"].str.split(';',expand=True)
iris_dataframe_csv

0    1    2    3          4
0    5.1  3.5  1.4  0.2     Setosa
1    4.9    3  1.4  0.2     Setosa
2    4.7  3.2  1.3  0.2     Setosa
3    4.6  3.1  1.5  0.2     Setosa
4      5  3.6  1.4  0.2     Setosa
..   ...  ...  ...  ...        ...
145  6.7    3  5.2  2.3  Virginica
146  6.3  2.5    5  1.9  Virginica
147  6.5    3  5.2    2  Virginica
148  6.2  3.4  5.4  2.3  Virginica
149  5.9    3  5.1  1.8  Virginica

[150 rows x 5 columns]

In [388]:
print('Shape ', iris_dataframe_csv.shape)
print('Type :', type(iris_dataset['data']))
print('First three rows of data :', iris_dataset['data'][:3])

Shape  (150, 5)
Type : <class 'numpy.ndarray'>
First three rows of data : [[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]]


2. Write a Python program to view basic statistical details like percentile, mean, std etc. of iris data.

   Ahora contesta cuál es la media del largo de los petalos en centímetros redondado a dos dígitos.

In [331]:
print(np.mean(iris_dataset['data'][, axis=0)

[5.84333333 3.05733333 3.758      1.19933333]


## Ejercicios

1. Write a Python program to split the iris dataset into its attributes (X) and labels (y). The X variable contains the first four columns (i.e. attributes) and y contains the labels of the dataset

2. Write a Python program using Scikit-learn to split the iris dataset into 70% train data and 30% test data. Out of total 150 records, the training set will contain 120 records and the test set contains 30 of those records. Print both datasets. 

3. Write a Python program using Scikit-learn to convert Species columns in a numerical column of the iris dataframe. To encode this data map convert each value to a number. e.g. Iris-setosa:0, Iris-versicolor:1, and Iris-virginica:2. Now print the iris dataset into 80% train data and 20% test data. Out of total 150 records, the training set will contain 120 records and the test set contains 30 of those records. Print both datasets

4. Write a Python program using Scikit-learn to split the iris dataset into 70% train data and 30% test data. Out of total 150 records, the training set will contain 105 records and the test set contains 45 of those records. Predict the response for test dataset (SepalLengthCm, SepalWidthCm, PetalLengthCm, PetalWidthCm) using the K Nearest Neighbor Algorithm. Use 5 as number of neighbors.

5. Write a Python program using Scikit-learn to split the iris dataset into 80% train data and 20% test data. Out of total 150 records, the training set will contain 120 records and the test set contains 30 of those records. Train or fit the data into the model and calculate the accuracy of the model using the K Nearest Neighbor Algorithm.

6. Write a Python program using Scikit-learn to split the iris dataset into 80% train data and 20% test data. Out of total 150 records, the training set will contain 120 records and the test set contains 30 of those records. Train or fit the data into the model and using the K Nearest Neighbor Algorithm calculate the performance for different values of k. 

7. Write a Python program using Scikit-learn to split the iris dataset into 80% train data and 20% test data. Out of total 150 records, the training set will contain 120 records and the test set contains 30 of those records. Train or fit the data into the model and using the K Nearest Neighbor Algorithm and create a plot to present the performance for different values of k. 

8. Write a Python program using Scikit-learn to split the iris dataset into 80% train data and 20% test data. Out of total 150 records, the training set will contain 120 records and the test set contains 30 of those records. Train or fit the data into the model and using the K Nearest Neighbor Algorithm and create a plot of k values vs accuracy.